docs/src/main/asciidoc/mp/integrations/oci.adoc

```diff
@@ -29,80 +29,233 @@ include::{rootdir}/includes/mp.adoc[]
+
+You can also manually push logs to the custom log configuration. First, add the following dependency to the project's ``pom.xml`` file:
+
+[source,xml]
+.Adding the dependency for Logging Ingestion API (to manually push logs)
+----
+<dependency>
+  <groupId>com.oracle.oci.sdk</groupId>
+  <artifactId>oci-java-sdk-loggingingestion</artifactId>
+</dependency>
+----
+
+Then, find the https://docs.oracle.com/en-us/iaas/Content/GSG/Tasks/contactingsupport_topic-Locating_Oracle_Cloud_Infrastructure_IDs.htm#resource-ocid[Oracle Cloud Identifier (OCID)] of the custom log configuration and use that OCID with the https://docs.oracle.com/en-us/iaas/api/#/en/logging-dataplane/20200831/[Logging Ingestion API] from the OCI SDK. See https://docs.oracle.com/en-us/iaas/Content/Logging/Concepts/using_the_api_customlogs.htm#using_the_api_customlogs[Ingesting Custom Logs] in OCI documentation.
+
+=== Tracing
+
+The OCI Application Performance Monitoring (APM) service provides deep visibility into the performance of applications and provides the ability to diagnose issues quickly in order to deliver a consistent level of service. See link:{oci-doc-url}/application-performance-monitoring/doc/application-performance-monitoring.html[About APM] in OCI documentation.
+
+Use OCI APM with Helidon to trace requests from start to finish, allowing you to analyze requests across your distributed services and thereby obtain a complete picture to identify bottlenecks and issues early on.
+
+To enable OCI Tracing integration, start by creating an APM Domain, an OCI resource that contains the systems being monitored by APM. See link:{oci-doc-url}/application-performance-monitoring/doc/create-apm-domain.html#GUID-ABC79A90-3940-4360-9E21-57D25B86F92B[Create an APM Domain] in OCI documentation.
+
+Make sure that you record the *Data Upload Endpoint* and *Data Keys* for the APM domain - you will need them to configure the APM Java Tracer on the application. If you need to find them again, follow the steps at link:{oci-doc-url}/application-performance-monitoring/doc/obtain-data-upload-endpoint-and-data-keys.html[Obtain Data Upload Endpoint and Data Keys] in OCI documentation.
+
+Next, configure the APM Java Tracer to use with Helidon. The APM Java Tracer (or just APM Tracer) is an APM data source that gathers and uploads data such as metrics and spans for monitoring. See link:{oci-doc-url}/application-performance-monitoring/doc/use-apm-tracer-helidon.html[Use APM Tracer in Helidon] in OCI documentation.
+
+After you configure OCI APM with Helidon, you can use OCI Trace Explorer to view traces and spans and identify performance issues in your monitored application. See link:{oci-doc-url}/application-performance-monitoring/doc/monitor-traces-trace-explorer.html#GUID-D23BCE18-F42F-44A1-B583-47DF9F9817D8[Monitor Traces in Trace Explorer] in OCI documentation.
+
+=== Streaming
```

Comment:
I only included the content from the OU course. Would you like me add the content from https://medium.com/helidon/helidon-messaging-with-oracle-streaming-service-68e0ef423853 as well

Reply:
I believe the OU material, and the Medium blog are complimentary so, if there is data in the blog that is not covered in the OU course, we should include it. Also, we want to be sure that the material is all refreshed for the current release of Helidon.